frontend/src/metabase/dashboard/reducers-typed.ts

```diff
@@ -413,3 +408,48 @@ export const loadingDashCards = createReducer(
       }));
   },
 );
+
+export const dashcardData = createReducer(
+  INITIAL_DASHBOARD_STATE.dashcardData,
+  builder => {
+    builder
+      .addCase(initialize, (state, action) => {
+        const { clearCache = true } = action.payload ?? {};
+        return clearCache ? {} : state;
+      })
+      .addCase(fetchCardDataAction.fulfilled, (state, action) => {
+        const { dashcard_id, card_id, result } = action.payload ?? {};
+        if (dashcard_id && card_id) {
+          return assocIn(state, [dashcard_id, card_id], result);
+        }
+      })
+      .addCase(clearCardData, (state, action) => {
+        const { cardId, dashcardId } = action.payload;
+        return dissocIn(state, [dashcardId, cardId]);
+      })
+      .addCase<string, { type: string; payload: { object: Card } }>(
+        Questions.actionTypes.UPDATE,
+        (state, action) => {
+          const id = action.payload.object.id;
+          for (const dashcardId in state) {
+            if (id in state[dashcardId]) {
+              delete state[dashcardId][id];
+            }
+          }
+        },
+      )
+      .addCase<string, { type: string; payload: Revision }>(
+        Revisions.actionTypes.REVERT,
+        (state, action) => {
+          const { model_id } = action.payload;
+          if (model_id) {
+            for (const dashcardId in state) {
+              if (model_id in state[dashcardId]) {
+                delete state[dashcardId][model_id];
+              }
```

Comment:
Same as the comment above. I believe this would simplify the code a little bit.

```suggestion
              delete state[dashcardId][model_id];
```